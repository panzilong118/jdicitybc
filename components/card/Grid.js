import * as React from 'react';

import PropTypes from 'prop-types';

import classNames from 'classnames';

import omit from 'omit.js';

export const CardGridProps = {
    prefixCls: PropTypes.string,
    style: PropTypes.object,
    className: PropTypes.string
};

function CardGrid(props) {
    const { prefixCls = 'jc-card', className = '', ...others } = props;
    const classString = classNames(`${prefixCls}-grid`, className);
    const otherProps = omit(others, ['prefixCls']);
    return <div {...otherProps} className={classString} />;
}

CardGrid.propTypes = { ...CardGridProps };

export default CardGrid;
