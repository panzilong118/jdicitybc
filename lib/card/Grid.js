'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.CardGridProps = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _omit = require('omit.js');

var _omit2 = _interopRequireDefault(_omit);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var CardGridProps = exports.CardGridProps = {
    prefixCls: _propTypes2.default.string,
    style: _propTypes2.default.object,
    className: _propTypes2.default.string
};

function CardGrid(props) {
    var _props$prefixCls = props.prefixCls,
        prefixCls = _props$prefixCls === undefined ? 'jc-card' : _props$prefixCls,
        _props$className = props.className,
        className = _props$className === undefined ? '' : _props$className,
        others = _objectWithoutProperties(props, ['prefixCls', 'className']);

    var classString = (0, _classnames2.default)(prefixCls + '-grid', className);
    var otherProps = (0, _omit2.default)(others, ['prefixCls']);
    return React.createElement('div', _extends({}, otherProps, { className: classString }));
}

CardGrid.propTypes = _extends({}, CardGridProps);

exports.default = CardGrid;
//# sourceMappingURL=Grid.js.map