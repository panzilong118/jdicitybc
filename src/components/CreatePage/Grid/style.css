.buttonBox{
    width: 100%;
    height: 50px;
    padding: 0 20px;
    border: 1px solid #ddd;
    line-height: 50px;
    background: #f7f7f7;
}