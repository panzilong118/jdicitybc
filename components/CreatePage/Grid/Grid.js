'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.default = undefined;

var _table = require('jdcloudui/lib/table');

var _table2 = _interopRequireDefault(_table);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

require('jdcloudui/lib/table/style');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./style.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Grid = function (_Component) {
    (0, _inherits3.default)(Grid, _Component);

    function Grid(props) {
        (0, _classCallCheck3.default)(this, Grid);

        var _this = (0, _possibleConstructorReturn3.default)(this, (Grid.__proto__ || (0, _getPrototypeOf2.default)(Grid)).call(this, props));

        _this.createColumns = function (columns) {
            var result = [];
            for (var i = 0; i < (columns || []).length; i++) {
                var temp = {};
                try {
                    temp.title = columns[i].title;
                    temp.key = columns[i].key;
                    temp.dataIndex = columns[i].dataIndex;
                    temp.render = columns[i].render;
                    temp.sorter = columns[i].sort || null;
                } catch (err) {
                    throw new Error('参数错误');
                }
                result.push(temp);
            }
            return result;
        };

        _this.createPagination = function (page) {
            if (!page) return false;
            var current = _this.props.dataSource && _this.props.dataSource.pageNum;
            var total = _this.props.dataSource && _this.props.dataSource.totalCount;
            return {
                current: current,
                total: total,
                showQuickJumper: true,
                pageSize: 10,
                onChange: _this.onChange.bind(_this)
            };
        };

        _this.onChange = function (pagination) {
            var search = _this.props.search;

            var params = (0, _extends3.default)({}, search, {
                pageNum: pagination
            });
            typeof _this.props.onChangeSearch === "function" && _this.props.onChangeSearch(params);
            _this.props.onSubmit(params);
        };

        return _this;
    }
    //创建Clums

    //创建分页

    //分页的change方法


    (0, _createClass3.default)(Grid, [{
        key: 'render',
        value: function render() {
            var _props = this.props,
                columns = _props.columns,
                dataSource = _props.dataSource,
                page = _props.page,
                rowSelection = _props.rowSelection,
                rowKey = _props.rowKey,
                options = _props.options;

            return _react2.default.createElement(
                'div',
                null,
                options && options.length && _react2.default.createElement(
                    'div',
                    { className: 'buttonBox' },
                    options.map(function (item) {
                        return item;
                    })
                ),
                _react2.default.createElement(_table2.default, {
                    columns: this.createColumns(columns),
                    dataSource: dataSource && dataSource.result,
                    pagination: this.createPagination(page),
                    rowSelection: rowSelection || null,
                    rowKey: rowKey || 'key'
                })
            );
        }
    }]);
    return Grid;
}(_react.Component);

exports.default = Grid;
module.exports = exports['default'];
//# sourceMappingURL=Grid.js.map