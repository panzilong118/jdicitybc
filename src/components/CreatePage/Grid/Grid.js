import React, { Component } from 'react';
import {Table, message} from 'jdcloudui';
import './style.css';

export default class Grid extends Component {
    constructor(props) {
        super(props);
    }
    //创建Clums
    createColumns = (columns) => {
        const result = [];
        for(let i = 0 ; i < (columns || []).length ; i++){
            let temp = {};
            try{
                temp.title = columns[i].title;
                temp.key = columns[i].key;
                temp.dataIndex = columns[i].dataIndex;
                temp.render = columns[i].render;
                temp.sorter = columns[i].sort || null;
            }catch(err){
                throw new Error('参数错误');
            }
            result.push(temp);
        }
        return result;
    }
    //创建分页
    createPagination = (page) => {
        if(!page) return false;
        let current = this.props.dataSource && this.props.dataSource.pageNum;
        let total = this.props.dataSource && this.props.dataSource.totalCount;
        return {
            current: current,
            total: total,
            showQuickJumper: true,
            pageSize: 10,
            onChange: this.onChange.bind(this)
        };
    }
    //分页的change方法
    onChange = (pagination) => {
        const {search} = this.props;
        const params = {
            ...search,
            pageNum: pagination,
        }
        typeof this.props.onChangeSearch === "function" && this.props.onChangeSearch(params);
        this.props.onSubmit(params);
    }
    render() {
        const {columns, dataSource, page, rowSelection, rowKey, options} = this.props;
        return (
            <div>
                {options && options.length &&
                    <div className="buttonBox">
                        {options.map((item) => {
                            return item;
                        })}
                    </div>
                }
                <Table
                    columns={this.createColumns(columns)}
                    dataSource={dataSource && dataSource.result}
                    pagination={this.createPagination(page)}
                    rowSelection={rowSelection || null}
                    rowKey={rowKey || 'key'}
                />
            </div>
        )
    }
}