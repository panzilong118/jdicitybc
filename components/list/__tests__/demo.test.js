import * as React from "react";

import renderer from 'react-test-renderer';

import Basic from '../demo/Basic';

import Simple from '../demo/Simple';

import LoadMoreList from '../demo/Loadmore';

import Vertical from '../demo/Vertical';

import Grid from '../demo/Grid';

import Responsive from '../demo/Resposive';

import InfiniteListExample from '../demo/Infinite-load';

import VirtualizedExample from '../demo/Infinite-virtualized-load';

describe('<List />', () => {

    it('Basic usage of List', () => {
        const demo = renderer.create(<Basic />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Simple usage of List', () => {
        const demo = renderer.create(<Simple />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('LoadMoreList usage of List', () => {
        const demo = renderer.create(<LoadMoreList />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Vertical usage of List', () => {
        const demo = renderer.create(<Vertical />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    // it('Grid usage of List', () => {
    //     const demo = renderer.create(<Grid />);
    //     let snapsDemo = demo.toJSON();
    //     expect(snapsDemo).toMatchSnapshot();
    // });
    //
    // it('Responsive usage of List', () => {
    //     const demo = renderer.create(<Responsive />);
    //     let snapsDemo = demo.toJSON();
    //     expect(snapsDemo).toMatchSnapshot();
    // });
    //
    // it('InfiniteListExample usage of List', () => {
    //     const demo = renderer.create(<InfiniteListExample />);
    //     let snapsDemo = demo.toJSON();
    //     expect(snapsDemo).toMatchSnapshot();
    // });

    it('VirtualizedExample usage of List', () => {
        const demo = renderer.create(<VirtualizedExample />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

});
