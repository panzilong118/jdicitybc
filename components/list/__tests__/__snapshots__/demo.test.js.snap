// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Basic usage of List 1`] = `
<div
  className="jc-list jc-list-split"
>
  <div>
    <div
      className="jc-spin-nested-loading"
      style={null}
    >
      <div
        className="jc-spin-container"
      >
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-meta"
          >
            <div
              className="jc-list-item-meta-avatar"
            >
              <span
                className="jc-avatar jc-avatar-circle jc-avatar-image"
              >
                <img
                  alt=""
                  onError={[Function]}
                  src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                />
              </span>
            </div>
            <div
              className="jc-list-item-meta-content"
            >
              <h4
                className="jc-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  Ant Design Title 1
                </a>
              </h4>
              <div
                className="jc-list-item-meta-description"
              >
                Ant Design, a design language for background applications, is refined by Ant UED Team
              </div>
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-meta"
          >
            <div
              className="jc-list-item-meta-avatar"
            >
              <span
                className="jc-avatar jc-avatar-circle jc-avatar-image"
              >
                <img
                  alt=""
                  onError={[Function]}
                  src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                />
              </span>
            </div>
            <div
              className="jc-list-item-meta-content"
            >
              <h4
                className="jc-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  Ant Design Title 2
                </a>
              </h4>
              <div
                className="jc-list-item-meta-description"
              >
                Ant Design, a design language for background applications, is refined by Ant UED Team
              </div>
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-meta"
          >
            <div
              className="jc-list-item-meta-avatar"
            >
              <span
                className="jc-avatar jc-avatar-circle jc-avatar-image"
              >
                <img
                  alt=""
                  onError={[Function]}
                  src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                />
              </span>
            </div>
            <div
              className="jc-list-item-meta-content"
            >
              <h4
                className="jc-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  Ant Design Title 3
                </a>
              </h4>
              <div
                className="jc-list-item-meta-description"
              >
                Ant Design, a design language for background applications, is refined by Ant UED Team
              </div>
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-meta"
          >
            <div
              className="jc-list-item-meta-avatar"
            >
              <span
                className="jc-avatar jc-avatar-circle jc-avatar-image"
              >
                <img
                  alt=""
                  onError={[Function]}
                  src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                />
              </span>
            </div>
            <div
              className="jc-list-item-meta-content"
            >
              <h4
                className="jc-list-item-meta-title"
              >
                <a
                  href="https://ant.design"
                >
                  Ant Design Title 4
                </a>
              </h4>
              <div
                className="jc-list-item-meta-description"
              >
                Ant Design, a design language for background applications, is refined by Ant UED Team
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<List /> LoadMoreList usage of List 1`] = `
<div
  className="jc-list demo-loadmore-list jc-list-split jc-list-loading jc-list-something-after-last-item"
>
  <div>
    <div
      className="jc-spin-nested-loading"
      style={null}
    >
      <div>
        <div
          className="jc-spin jc-spin-spinning jc-spin-show-text"
        >
          <span
            className="jc-spin-dot jc-spin-dot-spin"
          >
            <i />
            <i />
            <i />
            <i />
          </span>
        </div>
      </div>
      <div
        className="jc-spin-container jc-spin-blur"
      >
        <div
          style={
            Object {
              "minHeight": 53,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "height": 32,
          "lineHeight": "32px",
          "marginTop": 12,
          "textAlign": "center",
        }
      }
    >
      <button
        className="jc-btn"
        onClick={[Function]}
        type="button"
      >
        <span>
          loading more
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<List /> Simple usage of List 1`] = `
<div>
  <h3
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    Default Size
  </h3>
  <div
    className="jc-list jc-list-split jc-list-bordered jc-list-something-after-last-item"
  >
    <div
      className="jc-list-header"
    >
      <div>
        Header
      </div>
    </div>
    <div>
      <div
        className="jc-spin-nested-loading"
        style={null}
      >
        <div
          className="jc-spin-container"
        >
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Racing car sprays burning fuel into crowd.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Japanese princess to wed commoner.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Australian walks 100km after outback crash.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Man charged over missing wedding girl.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Los Angeles battles huge wildfires.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="jc-list-footer"
    >
      <div>
        Footer
      </div>
    </div>
  </div>
  <h3
    style={
      Object {
        "margin": "16px 0",
      }
    }
  >
    Small Size
  </h3>
  <div
    className="jc-list jc-list-sm jc-list-split jc-list-bordered jc-list-something-after-last-item"
  >
    <div
      className="jc-list-header"
    >
      <div>
        Header
      </div>
    </div>
    <div>
      <div
        className="jc-spin-nested-loading"
        style={null}
      >
        <div
          className="jc-spin-container"
        >
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Racing car sprays burning fuel into crowd.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Japanese princess to wed commoner.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Australian walks 100km after outback crash.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Man charged over missing wedding girl.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Los Angeles battles huge wildfires.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="jc-list-footer"
    >
      <div>
        Footer
      </div>
    </div>
  </div>
  <h3
    style={
      Object {
        "margin": "16px 0",
      }
    }
  >
    Large Size
  </h3>
  <div
    className="jc-list jc-list-lg jc-list-split jc-list-bordered jc-list-something-after-last-item"
  >
    <div
      className="jc-list-header"
    >
      <div>
        Header
      </div>
    </div>
    <div>
      <div
        className="jc-spin-nested-loading"
        style={null}
      >
        <div
          className="jc-spin-container"
        >
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Racing car sprays burning fuel into crowd.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Japanese princess to wed commoner.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Australian walks 100km after outback crash.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Man charged over missing wedding girl.
            </div>
          </div>
          <div
            className="jc-list-item"
          >
            <div
              className="jc-list-item-content jc-list-item-content-single"
            >
              Los Angeles battles huge wildfires.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="jc-list-footer"
    >
      <div>
        Footer
      </div>
    </div>
  </div>
</div>
`;

exports[`<List /> Vertical usage of List 1`] = `
<div
  className="jc-list jc-list-vertical jc-list-lg jc-list-split jc-list-something-after-last-item"
>
  <div>
    <div
      className="jc-spin-nested-loading"
      style={null}
    >
      <div
        className="jc-spin-container"
      >
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-extra-wrap"
          >
            <div
              className="jc-list-item-main"
            >
              <div
                className="jc-list-item-meta"
              >
                <div
                  className="jc-list-item-meta-avatar"
                >
                  <span
                    className="jc-avatar jc-avatar-circle jc-avatar-image"
                  >
                    <img
                      alt=""
                      onError={[Function]}
                      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                    />
                  </span>
                </div>
                <div
                  className="jc-list-item-meta-content"
                >
                  <h4
                    className="jc-list-item-meta-title"
                  >
                    <a
                      href="http://ant.design"
                    >
                      ant design part 0
                    </a>
                  </h4>
                  <div
                    className="jc-list-item-meta-description"
                  >
                    Ant Design, a design language for background applications, is refined by Ant UED Team.
                  </div>
                </div>
              </div>
              <div
                className="jc-list-item-content"
              >
                We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
              </div>
              <ul
                className="jc-list-item-action"
              >
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-star-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-like-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-message "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    2
                  </span>
                </li>
              </ul>
            </div>
            <div
              className="jc-list-item-extra"
            >
              <img
                alt="logo"
                src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                width={272}
              />
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-extra-wrap"
          >
            <div
              className="jc-list-item-main"
            >
              <div
                className="jc-list-item-meta"
              >
                <div
                  className="jc-list-item-meta-avatar"
                >
                  <span
                    className="jc-avatar jc-avatar-circle jc-avatar-image"
                  >
                    <img
                      alt=""
                      onError={[Function]}
                      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                    />
                  </span>
                </div>
                <div
                  className="jc-list-item-meta-content"
                >
                  <h4
                    className="jc-list-item-meta-title"
                  >
                    <a
                      href="http://ant.design"
                    >
                      ant design part 1
                    </a>
                  </h4>
                  <div
                    className="jc-list-item-meta-description"
                  >
                    Ant Design, a design language for background applications, is refined by Ant UED Team.
                  </div>
                </div>
              </div>
              <div
                className="jc-list-item-content"
              >
                We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
              </div>
              <ul
                className="jc-list-item-action"
              >
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-star-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-like-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-message "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    2
                  </span>
                </li>
              </ul>
            </div>
            <div
              className="jc-list-item-extra"
            >
              <img
                alt="logo"
                src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                width={272}
              />
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-extra-wrap"
          >
            <div
              className="jc-list-item-main"
            >
              <div
                className="jc-list-item-meta"
              >
                <div
                  className="jc-list-item-meta-avatar"
                >
                  <span
                    className="jc-avatar jc-avatar-circle jc-avatar-image"
                  >
                    <img
                      alt=""
                      onError={[Function]}
                      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                    />
                  </span>
                </div>
                <div
                  className="jc-list-item-meta-content"
                >
                  <h4
                    className="jc-list-item-meta-title"
                  >
                    <a
                      href="http://ant.design"
                    >
                      ant design part 2
                    </a>
                  </h4>
                  <div
                    className="jc-list-item-meta-description"
                  >
                    Ant Design, a design language for background applications, is refined by Ant UED Team.
                  </div>
                </div>
              </div>
              <div
                className="jc-list-item-content"
              >
                We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
              </div>
              <ul
                className="jc-list-item-action"
              >
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-star-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-like-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-message "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    2
                  </span>
                </li>
              </ul>
            </div>
            <div
              className="jc-list-item-extra"
            >
              <img
                alt="logo"
                src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                width={272}
              />
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-extra-wrap"
          >
            <div
              className="jc-list-item-main"
            >
              <div
                className="jc-list-item-meta"
              >
                <div
                  className="jc-list-item-meta-avatar"
                >
                  <span
                    className="jc-avatar jc-avatar-circle jc-avatar-image"
                  >
                    <img
                      alt=""
                      onError={[Function]}
                      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                    />
                  </span>
                </div>
                <div
                  className="jc-list-item-meta-content"
                >
                  <h4
                    className="jc-list-item-meta-title"
                  >
                    <a
                      href="http://ant.design"
                    >
                      ant design part 3
                    </a>
                  </h4>
                  <div
                    className="jc-list-item-meta-description"
                  >
                    Ant Design, a design language for background applications, is refined by Ant UED Team.
                  </div>
                </div>
              </div>
              <div
                className="jc-list-item-content"
              >
                We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
              </div>
              <ul
                className="jc-list-item-action"
              >
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-star-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-like-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-message "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    2
                  </span>
                </li>
              </ul>
            </div>
            <div
              className="jc-list-item-extra"
            >
              <img
                alt="logo"
                src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                width={272}
              />
            </div>
          </div>
        </div>
        <div
          className="jc-list-item"
        >
          <div
            className="jc-list-item-extra-wrap"
          >
            <div
              className="jc-list-item-main"
            >
              <div
                className="jc-list-item-meta"
              >
                <div
                  className="jc-list-item-meta-avatar"
                >
                  <span
                    className="jc-avatar jc-avatar-circle jc-avatar-image"
                  >
                    <img
                      alt=""
                      onError={[Function]}
                      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
                    />
                  </span>
                </div>
                <div
                  className="jc-list-item-meta-content"
                >
                  <h4
                    className="jc-list-item-meta-title"
                  >
                    <a
                      href="http://ant.design"
                    >
                      ant design part 4
                    </a>
                  </h4>
                  <div
                    className="jc-list-item-meta-description"
                  >
                    Ant Design, a design language for background applications, is refined by Ant UED Team.
                  </div>
                </div>
              </div>
              <div
                className="jc-list-item-content"
              >
                We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
              </div>
              <ul
                className="jc-list-item-action"
              >
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-star-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-like-o "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    156
                  </span>
                  <em
                    className="jc-list-item-action-split"
                  />
                </li>
                <li>
                  <span>
                    <i
                      className="jcicon jcicon-message "
                      style={
                        Object {
                          "marginRight": 8,
                        }
                      }
                    />
                    2
                  </span>
                </li>
              </ul>
            </div>
            <div
              className="jc-list-item-extra"
            >
              <img
                alt="logo"
                src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                width={272}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="jc-list-pagination"
    >
      <ul
        className="jc-pagination "
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="jc-pagination-disabled jc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <a
            className="jc-pagination-item-link"
          />
        </li>
        <li
          className="jc-pagination-item jc-pagination-item-1 jc-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="jc-pagination-disabled jc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <a
            className="jc-pagination-item-link"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<List /> VirtualizedExample usage of List 1`] = `
<div
  className="jc-list jc-list-split"
>
  <div>
    <div
      className="jc-spin jc-spin-show-text"
    >
      <span
        className="jc-spin-dot jc-spin-dot-spin"
      >
        <i />
        <i />
        <i />
        <i />
      </span>
    </div>
  </div>
</div>
`;
